src/asset/asset.js

```diff
@@ -70,89 +70,87 @@ var VARIANT_DEFAULT_PRIORITY = ['pvr', 'dxt', 'etc2', 'etc1', 'basis'];
  * @property {boolean} loading True if the resource is currently being loaded
  * @property {pc.AssetRegistry} registry The asset registry that this Asset belongs to
  */
-function Asset(name, type, file, data, options) {
-    EventHandler.call(this);
+class Asset extends EventHandler {
+    constructor(name, type, file, data, options) {
+        super();
 
-    this._id = assetIdCounter--;
+        this._id = assetIdCounter--;
 
-    this.name = name || '';
-    this.type = type;
-    this.tags = new Tags(this);
-    this._preload = false;
+        this.name = name || '';
```

Comment:
Use default parameter.

Reply:
Although having said that, the docs don't say the `name` parameter is optional. So maybe don't.